include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -344,9 +638,406 @@ class alignment_file_format_sam
+    {
+        typename target_type::value_type tmp;
+        read_field(std::forward<stream_view_type>(stream_view), tmp);
+        target = tmp;
+    }
+
+    /*!\brief Reads a list of values separated by comma as is the case for SAM tag arrays.
+     * \tparam variant_type     A specialization of std::variant that is stored in the seqan3::sam_tag_dictionary.
+     * \tparam stream_view_type The type of the stream as a view.
+     * \tparam value_type       The type of values to be stored in the tag array.
+     *
+     * \param[in, out] variant      A std::variant object to store the tag arrays in.
+     * \param[in, out] stream_view  The stream view to iterate over.
+     * \param[in]      value        A temporary value that determines the underlying type of the tag array.
+     *
+     * \details
+     *
+     * Reading the tags is done according to the official
+     * [SAM format specifications](https://samtools.github.io/hts-specs/SAMv1.pdf).
+     *
+     * The function throws a seqan3::parse_error if any unknown tag type was encountered. It will also fail if the
+     * format is not in a correct state (e.g. required fields are not giving), but throwing might occur downstream of
+     * the actual error.
+     */
+    template <typename variant_type, typename stream_view_type, typename value_type>
+    void read_sam_dict_vector(variant_type & variant, stream_view_type && stream_view, value_type value)
+    {
+        std::vector<value_type> tmp_vector;
+        while (begin(stream_view) != ranges::end(stream_view)) // not fully consumed yet
+        {
+            read_field(stream_view | view::take_until(is_char<','>), value);
+            tmp_vector.push_back(value);
+
+            if (is_char<','>(*begin(stream_view)))
+                ++begin(stream_view); // skip ','
+        }
+        variant = tmp_vector;
+    }
+
+    /*!\brief Reads the optional tag fields into the seqan3::sam_tag_dictionry.
+     * \tparam stream_view_type   The type of the stream as a view.
+     *
+     * \param[in, out] stream_view  The stream view to iterate over.
+     * \param[in, out] target       The seqan3::sam_tag_dictionary to store the tag information.
+     *
+     * \throws seqan3::parse_error if any unexpected character or format is encountered.
+     *
+     * \details
+     *
+     * Reading the tags is done according to the official
+     * [SAM format specifications](https://samtools.github.io/hts-specs/SAMv1.pdf).
+     *
+     * The function throws a seqan3::parse_error if any unknown tag type was encountered. It will also fail if the
+     * format is not in a correct state (e.g. required fields are not giving), but throwing might occur downstream of
+     * the actual error.
+     */
+    template <typename stream_view_type>
+    void read_field(stream_view_type && stream_view, sam_tag_dictionary & target)
+    {
+        uint16_t tag = static_cast<uint16_t>(*begin(stream_view)) * 256;
```

Comment:
```suggestion
        uint16_t tag = static_cast<uint16_t>(*begin(stream_view)) << 8;
```
Should be the same, but maybe it highlights the intention that we want to compact 2 8 bit integers into one 16 bit integer (if I remember correctly).